src/authentication/configuration.ts

```diff
@@ -0,0 +1,53 @@
+import * as vscode from 'vscode';
+
+export interface IHostConfiguration {
+	host: string;
+	username: string | undefined;
+	token: string | undefined;
+}
+
+export const HostHelper = class {
+	public static getApiHost(host: IHostConfiguration | vscode.Uri): vscode.Uri {
+		const hostUri: vscode.Uri = host instanceof vscode.Uri ? host : vscode.Uri.parse(host.host);
+		if (hostUri.authority === 'github.com') {
```

Comment:
is it possible for there to be a case mismatch?

Reply:
hmmmmm, good question. possibly, yes, I can test that quick